crates/bevy_reflect/src/lib.rs

```diff
@@ -46,6 +46,30 @@ pub use type_uuid::*;
 pub use bevy_reflect_derive::*;
 pub use erased_serde;
 
+#[doc(hidden)]
+pub mod __macro_exports {
+    use crate::Uuid;
+
+    #[allow(clippy::unusual_byte_groupings)] // unusual byte grouping is meant to signal the relevant bits
+    pub const fn generate_composite_uuid(a: Uuid, b: Uuid) -> Uuid {
+        let mut new = [0; 16];
+        let mut i = 0;
+        while i < new.len() {
+            new[i] = a.as_bytes()[i] ^ b.as_bytes()[i];
+
+            i += 1;
+        }
```

Comment:
Small nit:
```suggestion
        for i in 0..new.len() {
            new[i] = a.as_bytes()[i] ^ b.as_bytes()[i];
        }
```

Reply:
That sadly won't work, because for can't be used in `const fn`s yet